src/pages/NewSensor/NewSensorStepThree.js

```diff
@@ -101,10 +101,13 @@ const dispatchToProps = dispatch => {
   const updateLogInterval = value => dispatch(NewSensorActions.updateLogInterval(value));
   const previousTab = () => dispatch(WizardActions.previousTab());
   const exit = () => dispatch(goBack());
-
-  // TODO Use thunks to send commands to the sensor and save to the database.
   const connectToSensor = async () => {
-    await new Promise(r => setTimeout(r, 3000));
+    dispatch(NewSensorActions.updateSensor)
+      .then(dispatch(NewSensorActions.saveSensor))
+      .then(dispatch(gotoSettings()))
+      .catch(reason => {
+        ToastAndroid.show(reason, ToastAndroid.LONG);
```

Comment:
Is this the line that's causing the native error? Reason is an instance of an `Error` (an object, which is mapped to a `ReadableNativeMap` in java and the func wants a string (mapped to a standard java string)

Reply:
yeah, that's it. Thanks